### PR TITLE
Adding archetype for Siddhi extensions

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**

### CODE REVIEW

File: siddhi-extension-archetype/siddhi-execution/src/main/resources/archetype-resources/component/pom.xml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                        </Import-Package>
+                        <DynamicImport-Package>*</DynamicImport-Package>
+                        <Include-Resource>
+                            META-INF=target/classes/META-INF
+                        </Include-Resource>
+                    </instructions>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.jacoco</groupId>
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-execution/src/main/resources/archetype-resources/component/pom.xml]: Please refer https://github.com/wso2-extensions/siddhi-execution-unique/blob/master/component/pom.xml#L70 when adding jacoco plugin

File: siddhi-extension-archetype/siddhi-execution/src/main/resources/archetype-resources/component/src/main/resources/log4j.properties [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-execution/src/main/resources/archetype-resources/component/src/main/resources/log4j.properties]: remove licence from here

File: siddhi-extension-archetype/siddhi-execution/src/main/resources/archetype-resources/component/src/test/resources/log4j.properties [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-execution/src/main/resources/archetype-resources/component/src/test/resources/log4j.properties]: remove licence from here

File: siddhi-extension-archetype/siddhi-execution/src/main/resources/archetype-resources/mkdocs.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+extra_javascript:
+- javascripts/extra.js
+site_favicon: images/favicon.ico
+extra:
+  logo: images/siddhi-logo-w.svg
+  palette:
+    primary: teal
+    accent: teal
+pages:
+- Welcome to WSO2 Siddhi IO File: index.md
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-execution/src/main/resources/archetype-resources/mkdocs.yml]: this line should be templated

File: siddhi-extension-archetype/siddhi-io/src/main/resources/archetype-resources/README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+## Latest API Docs
+
+
+## How to use
+
+**Using the extension in <a target="_blank" href="https://github.com/wso2/product-sp">WSO2 Stream Processor</a>**
+
+* You can use this extension in the latest <a target="_blank" href="https://github.com/wso2/product-sp/releases">WSO2 Stream Processor</a> that is a part of <a target="_blank" href="http://wso2.com/analytics?utm_source=gitanalytics&utm_campaign=gitanalytics_Jul17">WSO2 Analytics</a> offering, with editor, debugger and simulation support.
+
+* This extension is shipped by default with WSO2 Stream Processor, if you wish to use an alternative version of this extension you can replace the component <a target="_blank" href="https://github.com/wso2-extensions/siddhi-execution-unique/releases">jar</a> that can be found in the `<STREAM_PROCESSOR_HOME>/lib` directory.
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-io/src/main/resources/archetype-resources/README.md]: should be templated. Has execution-unique wording

File: siddhi-extension-archetype/siddhi-io/src/main/resources/archetype-resources/component/src/main/java/__typeOf_IO__/sink/__classNameOfSink__.java [hunk 6]
```diff
@@ -0,0 +1,168 @@
+package ${package}.${typeOf_IO}.sink;
+
+import org.apache.log4j.Logger;
+import org.wso2.siddhi.annotation.Example;
+import org.wso2.siddhi.annotation.Extension;
+/*import org.wso2.siddhi.annotation.Parameter;
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-io/src/main/resources/archetype-resources/component/src/main/java/__typeOf_IO__/sink/__classNameOfSink__.java]: comments??

File: siddhi-extension-archetype/siddhi-io/src/main/resources/archetype-resources/component/src/main/resources/log4j.properties [hunk 7]
```diff
@@ -0,0 +1,35 @@
+#
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-io/src/main/resources/archetype-resources/component/src/main/resources/log4j.properties]: remove licence

File: siddhi-extension-archetype/siddhi-io/src/main/resources/archetype-resources/component/src/test/resources/log4j.properties [hunk 8]
```diff
@@ -0,0 +1,31 @@
+#
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-io/src/main/resources/archetype-resources/component/src/test/resources/log4j.properties]: remove licence

File: siddhi-extension-archetype/siddhi-io/src/main/resources/archetype-resources/mkdocs.yml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+extra_javascript:
+- javascripts/extra.js
+site_favicon: images/favicon.ico
+extra:
+  logo: images/siddhi-logo-w.svg
+  palette:
+    primary: teal
+    accent: teal
+pages:
+- Welcome to WSO2 Siddhi IO File: index.md
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-io/src/main/resources/archetype-resources/mkdocs.yml]: Need to template this

File: siddhi-extension-archetype/siddhi-map/src/main/resources/archetype-resources/README.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+## Latest API Docs
+
+
+## How to use
+
+**Using the extension in <a target="_blank" href="https://github.com/wso2/product-sp">WSO2 Stream Processor</a>**
+
+* You can use this extension in the latest <a target="_blank" href="https://github.com/wso2/product-sp/releases">WSO2 Stream Processor</a> that is a part of <a target="_blank" href="http://wso2.com/analytics?utm_source=gitanalytics&utm_campaign=gitanalytics_Jul17">WSO2 Analytics</a> offering, with editor, debugger and simulation support.
+
+* This extension is shipped by default with WSO2 Stream Processor, if you wish to use an alternative version of this extension you can replace the component <a target="_blank" href="https://github.com/wso2-extensions/siddhi-execution-unique/releases">jar</a> that can be found in the `<STREAM_PROCESSOR_HOME>/lib` directory.
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-map/src/main/resources/archetype-resources/README.md]: Need to template

File: siddhi-extension-archetype/siddhi-map/src/main/resources/archetype-resources/component/src/main/resources/log4j.properties [hunk 11]
```diff
@@ -0,0 +1,35 @@
+#
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-map/src/main/resources/archetype-resources/component/src/main/resources/log4j.properties]: remove licence

File: siddhi-extension-archetype/siddhi-map/src/main/resources/archetype-resources/component/src/test/resources/log4j.properties [hunk 12]
```diff
@@ -0,0 +1,31 @@
+#
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-map/src/main/resources/archetype-resources/component/src/test/resources/log4j.properties]: remove licence

File: siddhi-extension-archetype/siddhi-map/src/main/resources/archetype-resources/mkdocs.yml [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+extra_javascript:
+- javascripts/extra.js
+site_favicon: images/favicon.ico
+extra:
+  logo: images/siddhi-logo-w.svg
+  palette:
+    primary: teal
+    accent: teal
+pages:
+- Welcome to WSO2 Siddhi IO File: index.md
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-map/src/main/resources/archetype-resources/mkdocs.yml]: please template name

File: siddhi-extension-archetype/siddhi-store/src/main/resources/archetype-resources/README.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+
+
+## Latest API Docs
+
+
+## How to use
+
+**Using the extension in <a target="_blank" href="https://github.com/wso2/product-sp">WSO2 Stream Processor</a>**
+
+* You can use this extension in the latest <a target="_blank" href="https://github.com/wso2/product-sp/releases">WSO2 Stream Processor</a> that is a part of <a target="_blank" href="http://wso2.com/analytics?utm_source=gitanalytics&utm_campaign=gitanalytics_Jul17">WSO2 Analytics</a> offering, with editor, debugger and simulation support.
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-store/src/main/resources/archetype-resources/README.md]: template

File: siddhi-extension-archetype/siddhi-store/src/main/resources/archetype-resources/component/src/main/resources/log4j.properties [hunk 15]
```diff
@@ -0,0 +1,35 @@
+#
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-store/src/main/resources/archetype-resources/component/src/main/resources/log4j.properties]: remove licence

File: siddhi-extension-archetype/siddhi-store/src/main/resources/archetype-resources/component/src/test/resources/log4j.properties [hunk 16]
```diff
@@ -0,0 +1,31 @@
+#
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-store/src/main/resources/archetype-resources/component/src/test/resources/log4j.properties]: remove licence

File: siddhi-extension-archetype/siddhi-store/src/main/resources/archetype-resources/mkdocs.yml [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+extra_javascript:
+- javascripts/extra.js
+site_favicon: images/favicon.ico
+extra:
+  logo: images/siddhi-logo-w.svg
+  palette:
+    primary: teal
+    accent: teal
+pages:
+- Welcome to WSO2 Siddhi IO File: index.md
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-store/src/main/resources/archetype-resources/mkdocs.yml]: please template

File: siddhi-extension-archetype/pom.xml [hunk 18]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (kalaiyarasiganeshalingam) [reply siddhi-extension-archetype/pom.xml]: Fixed in 	399ac86

File: siddhi-extension-archetype/siddhi-execution/pom.xml [hunk 19]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (kalaiyarasiganeshalingam) [reply siddhi-extension-archetype/siddhi-execution/pom.xml]: Fixed in 	399ac86

File: siddhi-extension-archetype/siddhi-store/src/main/resources/archetype-resources/README.md (hunk 14 above)
Comment (kalaiyarasiganeshalingam) [reply siddhi-extension-archetype/siddhi-store/src/main/resources/archetype-resources/README.md]: Fixed in 	399ac86

File: siddhi-extension-archetype/siddhi-execution/src/main/resources/archetype-resources/README.md [hunk 20]
```diff
@@ -1,4 +1,4 @@
-﻿siddhi-io-
+﻿siddhi-io-${executionType}
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-execution/src/main/resources/archetype-resources/README.md]: Isn't this for execution extensions? Then the above should be siddhi-execution-{executionType}

File: siddhi-extension-archetype/siddhi-io/src/main/resources/archetype-resources/README.md [hunk 21]
```diff
@@ -11,7 +11,7 @@
 
 * You can use this extension in the latest <a target="_blank" href="https://github.com/wso2/product-sp/releases">WSO2 Stream Processor</a> that is a part of <a target="_blank" href="http://wso2.com/analytics?utm_source=gitanalytics&utm_campaign=gitanalytics_Jul17">WSO2 Analytics</a> offering, with editor, debugger and simulation support.
 
-* This extension is shipped by default with WSO2 Stream Processor, if you wish to use an alternative version of this extension you can replace the component <a target="_blank" href="https://github.com/wso2-extensions/siddhi-execution-unique/releases">jar</a> that can be found in the `<STREAM_PROCESSOR_HOME>/lib` directory.
+* This extension is shipped by default with WSO2 Stream Processor, if you wish to use an alternative version of this extension you can replace the component <a target="_blank" href="https://github.com/wso2-extensions/siddhi-execution-${typeOf_IO}/releases">jar</a> that can be found in the `<STREAM_PROCESSOR_HOME>/lib` directory.
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-io/src/main/resources/archetype-resources/README.md]: URL should be fixed as siddhi-io-${typeOf_IO}

File: siddhi-extension-archetype/siddhi-store/src/main/resources/archetype-resources/README.md [hunk 22]
```diff
@@ -11,7 +11,7 @@
 
 * You can use this extension in the latest <a target="_blank" href="https://github.com/wso2/product-sp/releases">WSO2 Stream Processor</a> that is a part of <a target="_blank" href="http://wso2.com/analytics?utm_source=gitanalytics&utm_campaign=gitanalytics_Jul17">WSO2 Analytics</a> offering, with editor, debugger and simulation support.
 
-* This extension is shipped by default with WSO2 Stream Processor, if you wish to use an alternative version of this extension you can replace the component <a target="_blank" href="https://github.com/wso2-extensions/siddhi-execution-unique/releases">jar</a> that can be found in the `<STREAM_PROCESSOR_HOME>/lib` directory.
+* This extension is shipped by default with WSO2 Stream Processor, if you wish to use an alternative version of this extension you can replace the component <a target="_blank" href="https://github.com/wso2-extensions/siddhi-execution-${storeType}/releases">jar</a> that can be found in the `<STREAM_PROCESSOR_HOME>/lib` directory.
```
Comment (tishan89) [inline siddhi-extension-archetype/siddhi-store/src/main/resources/archetype-resources/README.md]: URL should be fixed

**Comment by kalaiyarasiganeshalingam** [comment]
I am closing this PR. Since siddhi-maven-archetype extension has created separately.

**Pull request closed without merging**